packages/expo-file-system/android/src/main/java/expo/modules/filesystem/ExpokitFilePermissionChecker.java

```diff
@@ -0,0 +1,58 @@
+package expo.modules.filesystem;
+
+import android.content.Context;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+
+import expo.core.interfaces.InternalModule;
+import main.FilePermissionChecker;
+import main.Permission;
+
+public class ExpokitFilePermissionChecker implements FilePermissionChecker, InternalModule {
+  @Override
+  public EnumSet<Permission> getInfo(Context context, String path) {
+    try {
+      path = new File(path).getCanonicalPath();
+      String filesDir = context.getFilesDir().getCanonicalPath();
+      if (path.startsWith(filesDir + "/")) {
+        return EnumSet.of(Permission.READ, Permission.WRITE);
+      }
+      if (filesDir.equals(path)) {
```

Comment:
Maybe we could somehow merge these two `if`s into one? Is there some directory unifying method that would ensure that `path` ends with `/` and `filesDir` ends with `/` so we can use only `path.startsWith(filesDir)`?